examples/training/semantic_segmentation/pascal_voc/basic_segmentation_training.py

```diff
@@ -0,0 +1,186 @@
+    "tensorboard_path", None, "Directory which will be used to store tensorboard logs."
+)
+flags.DEFINE_integer(
+    "batch_size",
+    4,
+    "Batch size for training and evaluation. This will be multiplied by the number of accelerators in use.",
+)
+flags.DEFINE_boolean(
+    "use_xla", True, "Whether or not to use XLA (jit_compile) for training."
+)
+flags.DEFINE_boolean(
+    "use_mixed_precision",
+    False,
+    "Whether or not to use FP16 mixed precision for training.",
+)
+flags.DEFINE_boolean(
+    "use_ema",
+    True,
+    "Whether or not to use exponential moving average weight updating",
+)
+flags.DEFINE_boolean(
+    "include_rescaling",
+    True,
+    "Whether or not to include rescaling in the model",
+)
+flags.DEFINE_string(
+    "backbone_weights",
+    "imagenet",
+    "Path for the weights to be loaded into the backbone",
+)
+flags.DEFINE_string(
+    "backbone",
+    None,
+    "String denoting a supported backbone for the segmentation model",
+)
+flags.DEFINE_float(
+    "initial_learning_rate",
+    0.007,
+    "Initial learning rate which will reduce on plateau. This will be multiplied by the number of accelerators in use",
+)
+flags.DEFINE_string(
+    "model_kwargs",
+    "{}",
+    "Keyword argument dictionary to pass to the constructor of the model being trained",
+)
+
+flags.DEFINE_float(
+    "weight_decay",
+    5e-4,
+    "Weight decay parameter for the optimizer",
+)
+
+# An upper bound for number of epochs (this script uses EarlyStopping).
+flags.DEFINE_integer("epochs", 1000, "Epochs to train for")
+
+FLAGS = flags.FLAGS
+FLAGS(sys.argv)
+
+# Try to detect an available TPU. If none is present, default to MirroredStrategy
+try:
+    tpu = tf.distribute.cluster_resolver.TPUClusterResolver.connect()
+    strategy = tf.distribute.TPUStrategy(tpu)
+    if FLAGS.mixed_precision:
+        logging.info("Mixed precision training enabled")
+        tf.keras.mixed_precision.set_global_policy("mixed_bfloat16")
+except ValueError:
+    # MirroredStrategy is best for a single machine with one or multiple GPUs
+    strategy = tf.distribute.MirroredStrategy()
+    if FLAGS.mixed_precision:
+        logging.info("Mixed precision training enabled")
+        tf.keras.mixed_precision.set_global_policy("mixed_float16")
+print("Number of accelerators: ", strategy.num_replicas_in_sync)
+
+local_batch = FLAGS.batch_size
+global_batch = local_batch * strategy.num_replicas_in_sync
+base_lr = FLAGS.initial_learning_rate * strategy.num_replicas_in_sync
+
+train_ds = load(split="sbd_train", data_dir=None)
+eval_ds = load(split="sbd_eval", data_dir=None)
+
+
+def preprocess_image(img, cls_seg):
+    img = tf.keras.layers.Resizing(512, 512, interpolation="nearest")(img)
+    inputs = {"images": img, "segmentation_masks": cls_seg}
+    inputs = keras_cv.layers.preprocessing.RandomFlip("horizontal")(inputs)
+    return inputs["images"], inputs["segmentation_masks"]
+
+
+def proc_train_fn(examples):
+    image = examples.pop("image")
```

Comment:
Maybe flip and then resize instead? 
Since we're getting a dict, popping, turning into a dict again to do `RandomFlip` and then returning separate outputs again.

The former might be more computationally expensive though, in which case this approach is better?